src/GovUk.Education.ExploreEducationStatistics.Data.Services/TableBuilderDataService.cs

```diff
@@ -1,45 +1,68 @@
+            IUserService userService,
+            IResultBuilder<Observation, ObservationViewModel> resultBuilder) :
+            base(observationService, subjectService, userService)
         {
             _resultBuilder = resultBuilder;
             _resultSubjectMetaService = resultSubjectMetaService;
+            _subjectService = subjectService;
         }
 
-        public override TableBuilderResultViewModel Query(ObservationQueryContext queryContext, ReleaseId? releaseId = null)
+        public override Task<Either<ActionResult, TableBuilderResultViewModel>> Query(
+            ObservationQueryContext queryContext)
         {
-            var observations = GetObservations(queryContext, releaseId).AsQueryable();
-            if (!observations.Any())
-            {
-                return new TableBuilderResultViewModel();
-            }
+            return CheckSubjectExists(queryContext.SubjectId)
+                .OnSuccess(CheckCanViewSubjectData)
+                .OnSuccess(_ =>
+                {
+                    var observations = GetObservations(queryContext).AsQueryable();
+                    if (!observations.Any())
+                    {
+                        return Task.FromResult(
+                            new Either<ActionResult, TableBuilderResultViewModel>(new TableBuilderResultViewModel()));
+                    }
 
-            var subjectMetaViewModel =
-                _resultSubjectMetaService.GetSubjectMeta(SubjectMetaQueryContext.FromObservationQueryContext(queryContext), observations);
+                    return _resultSubjectMetaService
+                        .GetSubjectMeta(SubjectMetaQueryContext.FromObservationQueryContext(queryContext), observations)
+                        .OnSuccess(subjectMetaViewModel =>
+                        {
+                            return new TableBuilderResultViewModel
+                            {
+                                SubjectMeta = subjectMetaViewModel,
+                                Results = observations.Select(observation =>
+                                    _resultBuilder.BuildResult(observation, queryContext.Indicators))
+                            };
+                        });
+                });
+        }
 
-            return new TableBuilderResultViewModel
-            {
-                SubjectMeta = subjectMetaViewModel,
-                Results = observations.Select(observation =>
-                    _resultBuilder.BuildResult(observation, queryContext.Indicators))
-            };
+        private async Task<Either<ActionResult, bool>> CheckCanViewSubjectData(Subject subject)
+        {
+            var result = _subjectService.IsSubjectForLatestPublishedRelease(subject.Id) || await _userService.MatchesPolicy(subject, CanViewSubjectData);
+            return result ? new Either<ActionResult, bool>(true) : new ForbidResult();
```

Comment:
Would the compiler not let you have `return result ? true : new ForbidResult();`?

Reply:
No with the ternary statement, the error is `There is no implicit conversion between bool and ForbidResult`. If can rewrite it without the statement just fine.

Reply:
Rewritten

Reply:
Ahh was happy the way it was, but happy this way too!  Ta